src/pages/Payments/Payments.tsx

```diff
@@ -0,0 +1,42 @@
+import { useStepper } from '@/hooks/useStepper';
+import { CardInfoProvider } from '@/context/CardInfo';
+import { OverlayProvider } from '@/context/Overlay';
+import AddCard from './stepper/AddCard';
+import AddCardCompleted from './stepper/AddCardCompleted';
+import CardList from './stepper/CardList';
+
+export type StepType = typeof CARD_REGISTRATION_STEPS;
+
+const CARD_REGISTRATION_STEPS = [
+  '카드_입력',
+  '카드_입력_완료',
+  '카드_목록',
+] as const;
+
+export default function Payments() {
+  const { Stepper, Step, setStep } = useStepper<StepType>(
+    CARD_REGISTRATION_STEPS
+  );
+
+  const onChangeStep = (step: 0 | 1 | 2) => () => {
+    setStep(CARD_REGISTRATION_STEPS[step]);
+  };
+
+  return (
+    <OverlayProvider>
+      <CardInfoProvider>
+        <Stepper>
+          <Step name={CARD_REGISTRATION_STEPS[0]}>
+            <AddCard onNext={onChangeStep(1)} onPrevious={onChangeStep(2)} />
+          </Step>
+          <Step name={CARD_REGISTRATION_STEPS[1]}>
+            <AddCardCompleted onNext={onChangeStep(2)} />
+          </Step>
+          <Step name={CARD_REGISTRATION_STEPS[2]}>
+            <CardList onNext={onChangeStep(0)} moveToStep={onChangeStep(1)} />
+          </Step>
+        </Stepper>
```

Comment:
```suggestion
            <Stepper>
          <Step name="카드_입력">
            <AddCard onNext={onChangeStep(1)} onPrevious={onChangeStep(2)} />
          </Step>
          <Step name="카드_입력_완료">
            <AddCardCompleted onNext={onChangeStep(2)} />
          </Step>
          <Step name="카드_입력_완료">
            <CardList onNext={onChangeStep(0)} moveToStep={onChangeStep(1)} />
          </Step>
        </Stepper>
```

이쁘게 제네릭으로 타입추론도 가능하도록 만들어놓은 만큼 TS의 장점을 좀 더 누려볼까요?
Step 컴포넌트의 name은 추론이 잘 되도록 구성되어있기때문에 할당되어있는 값보다 어떤 name인지 명확하게 보여줘도 문제없을 것 같아요!

Reply:
각 step의 이름을 더욱 명시적으로 나타낼 수 있겠네요!